subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan_aux.c

```diff
@@ -584,27 +587,142 @@ static int isr_rx_pdu(struct lll_scan_aux *lll, uint8_t devmatch_ok,
 
 		return 0;
 
+	/* Active scanner */
 	} else if (!lll_scan->conn &&
+		   lll_scan->type &&
+		   !lll->state &&
+		   (pdu->adv_ext_ind.adv_mode & BT_HCI_LE_ADV_PROP_SCAN) &&
 		   lll_scan_ext_tgta_check(lll_scan, false, false, pdu,
 					   rl_idx)) {
 #else /* !CONFIG_BT_CENTRAL */
-	} else if (lll_scan_ext_tgta_check(lll_scan, false, false, pdu,
+	} else if (lll_scan->type &&
+		   !lll->state &&
+		   (pdu->adv_ext_ind.adv_mode & BT_HCI_LE_ADV_PROP_SCAN) &&
+		   lll_scan_ext_tgta_check(lll_scan, false, false, pdu,
 					   rl_idx)) {
 #endif /* !CONFIG_BT_CENTRAL */
+		struct node_rx_pdu *rx;
+		struct pdu_adv *pdu_tx;
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+		bt_addr_t *lrpa;
+#endif /* CONFIG_BT_CTLR_PRIVACY */
+
+		/* Check if 4 nodes free, 2 will be utilized for aux PDU and
+		 * scan response PDU; 2 more to ensure connections have them.
+		 */
+		rx = ull_pdu_rx_alloc_peek(4);
+		if (!rx) {
+			return -ENOBUFS;
+		}
+
+		/* setup tIFS switching */
+		radio_tmr_tifs_set(EVENT_IFS_US);
+		radio_switch_complete_and_rx(lll->phy);
+
+		/* prepare the scan request packet */
+		pdu_tx = (void *)radio_pkt_scratch_get();
+		pdu_tx->type = PDU_ADV_TYPE_SCAN_REQ;
+		pdu_tx->rx_addr = pdu->tx_addr;
+		pdu_tx->len = sizeof(struct pdu_adv_scan_req);
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+		lrpa = ull_filter_lll_lrpa_get(rl_idx);
+		if (lll_scan->rpa_gen && lrpa) {
+			pdu_tx->tx_addr = 1;
+			memcpy(pdu_tx->scan_req.scan_addr, lrpa->val,
+			       BDADDR_SIZE);
+		} else {
+#else
+		if (1) {
+#endif /* CONFIG_BT_CTLR_PRIVACY */
+			pdu_tx->tx_addr = lll_scan->init_addr_type;
+			memcpy(pdu_tx->scan_req.scan_addr,
```

Comment:
Since you already started based on my previous comment ;) (applies multiple places). 

```suggestion
			(void)memcpy(pdu_tx->scan_req.scan_addr,
```